src/main/java/omero/gateway/model/DatasetData.java

```diff
@@ -187,7 +188,7 @@ public Map<Long, Long> getAnnotationsCounts() {
      *
      * @return See above.
      */
-    public Set getImages() {
+    public Set<ImageData> getImages() {
```

Comment:
This could break using the method

Reply:
Is there a non-breaking way to fix that? Although I'm not sure if it really would break client code... I'll test. I'll also add integration tests for the new class.

Reply:
when adding new tests, you will need to disable the GHA otherwise it will be red.

Reply:
thanks for checking